UnityProject/Assets/Scripts/Player/PlayerScript.cs

```diff
@@ -575,6 +590,82 @@ public void OnMouseExit()
 		UIManager.SetToolTip = "";
 	}
 
+	public System.Random RNG = new System.Random();
+
+	public void OnInteract(TargetedInteraction Interaction, Component interactable)
+	{
+		if (Interaction != null)
+		{
+			if (ComponentManager.TryGetUniversalObjectPhysics(interactable.gameObject, out var UOP))
+			{
+				var Detail = UOP.attributes.Component.OrNull()?.AppliedDetail;
+				if (Detail == null) return;
+
+				if (RNG.Next(0, 100) > 55)
+				{
+
+					bool WearingGloves = false;
+					var slots = DynamicItemStorage.GetNamedItemSlots(NamedSlot.hands);
+					if (slots.Count != 0)
```

Comment:
You can skip this check and simply test slot for `null`. That also allows the no gloves part to be written in an `else`.

Reply:
Are you sure? I checked the enum extensions and thought it could return null.

Reply:
is linq

Reply:
If [this](https://github.com/unitystation/unitystation/blob/develop/UnityProject/Assets/Scripts/Util/EnumerableExt.cs#L12) is the implementation used, it can return `null` when the list is empty.